[WIP] Decouple from suit and prepare for jet

AidanCurrah opened the pull request:
Relies on [PR for CMS](https://github.com/onespacemedia/cms/pull/153). Until both a ready do not merge.

This decouples the PT from suit and gets it ready for the migration to Jet. It uses django-adminsortable for the sortables instead and removes most of our admin template overrides.

Bugs
---

* [x] Section field hiding/showing does not work when first adding a section - it shows all fields
* [x] WYSIWYG cannot be focused when first adding a section
* [x] Update login page to not have `user and password: admin`
* [x] Sitemap module needs some styling (see below)
* [x] Sitemap module doesn't have ordering controls
* [x] Sitemap module is not draggable on the admin homepage
* [x] Disable save button on click as you can click multiple times to add a few pages at once
* [x] 404 on admin homepage: `(index):59 GET http://localhost:3000/static/admin/js/jquery.js 404 (Not Found)`
* [x] Exception on the 'Change file' page (traceback below)
* [x] Image editor is not loading on the change file page
* [x] Rearrange fieldsets on `ModelAdmin`s on the default apps so that the most important information (e.g. text on news articles) is shown on the first tab
* [x] Clicking the "+" next to the widget for an `ImageRefField` doesn't work - try adding a new image for a news article, for example
* [x] On that note, the edit link there doesn't work either (greyed out) 
* [x] Closing the edit screen modal (brought up by clicking 'Edit' on the front-end editing bar) raises an exception (so the modal never closes)
* [x] Cannot insert new images into the media library and insert them into a WYSIWYG at the same time (traceback in the comments)

Enhancements
---

* [x] Can we remove the delay when showing submenus in the admin?
* [x] Can we remove 'Application page' from submenus? (yep - with a custom menu and not supplying `app_label`)
* [x] It'd be neat if django-sortedm2m works with Jet!
* [x] Improve 'add page' screen

AidanCurrah committed:
Initial migration work, requires testing on non-news models. This removes the use of suit sortables from all admin models and removes most templates as their functionality is included in jet.

Commit:
Remove 'order' from FAQ and partner list_editable.

lewiscollard commented:
Updating original comment to add bug with sections JS (hiding/showing fields doesn't work when first adding a section).

Edit: Seems that the class names aren't what's expected. Existing items have the class `dynamic-contentsection_set` which matches the [selector `[class*=section_set]`](https://github.com/onespacemedia/project-template/blob/9e3e8465816c3849f16f243460f08d0808fb3055/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/apps/sections/templates/admin/pages/page/sections.js#L9) but newly-added ones do not.

lewiscollard commented:
Added two issues with the sitemap module to the checklist.

![screen shot 2018-08-15 at 12 00 52](https://user-images.githubusercontent.com/4973097/44145331-176eb05c-a083-11e8-912f-4035f8607081.png)

lewiscollard commented:
New item on checklist: Exception when saving the page when using the edit bar on the front end:

```
bundle.min.js:1 TypeError: Cannot read property 'find' of undefined
    at s.processPopupResponse (bundle.min.js:1)
    at bundle.min.js:1
    at s.closePopup (bundle.min.js:1)
    at HTMLDivElement.<anonymous> (bundle.min.js:1)
    at Function.each (bundle.min.js:8)
    at ot.fn.init.each (bundle.min.js:7)
    at s.findPopupResponse (bundle.min.js:1)
    at s.run (bundle.min.js:1)
    at HTMLDocument.<anonymous> (bundle.min.js:1)
    at u (bundle.min.js:8) "TypeError: Cannot read property 'find' of undefined
    at s.processPopupResponse (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21701)
    at http://localhost:3000/static/jet/js/build/bundle.min.js:1:21195
    at s.closePopup (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21432)
    at HTMLDivElement.<anonymous> (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21595)
    at Function.each (http://localhost:3000/static/jet/js/build/bundle.min.js:8:1086)
    at ot.fn.init.each (http://localhost:3000/static/jet/js/build/bundle.min.js:7:30940)
    at s.findPopupResponse (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21536)
    at s.run (http://localhost:3000/static/jet/js/build/bundle.min.js:1:22998)
    at HTMLDocument.<anonymous> (http://localhost:3000/static/jet/js/build/bundle.min.js:1:23182)
    at u (http://localhost:3000/static/jet/js/build/bundle.min.js:8:24846)"
run @ bundle.min.js:1
(anonymous) @ bundle.min.js:1
u @ bundle.min.js:8
fireWith @ bundle.min.js:8
ready @ bundle.min.js:8
a @ bundle.min.js:7
```

lewiscollard commented:
Added to checklist: Problem inserting images from the library into WYSIWYGs.

```
bundle.min.js?v=1.0.7:1 Uncaught TypeError: Cannot read property 'is' of undefined
    at s.processPopupResponse (bundle.min.js?v=1.0.7:1)
    at bundle.min.js?v=1.0.7:1
    at s.closePopup (bundle.min.js?v=1.0.7:1)
    at HTMLAnchorElement.<anonymous> (bundle.min.js?v=1.0.7:1)
    at HTMLAnchorElement.dispatch (bundle.min.js?v=1.0.7:9)
    at HTMLAnchorElement.m.handle (bundle.min.js?v=1.0.7:9)
```

Commit:
Remove 'use admin for username and password' from login screen.

lewiscollard commented:
Edited: 404 on admin homepage, split out some "would be nice".

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html:
```diff
@@ -1 +1,6 @@
 {% extends "admin/change_list.html" %}
+
+{# TODO Discuss this change in the CMS instead of overriding it in a template. #}
```
Does this template do anything at all?

AidanCurrah replied:
it's overwriting the CMS template this is:
```
{% extends "admin/change_list.html" %}


{% block object-tools-items %}
    <a class="btn" href="{% url 'admin:auth_user_invite' %}">Invite user</a>
    {{block.super}}
{% endblock %}
```
As the link was throwing errors and I didn't know if anyone ever used the user inviting feature.

lewiscollard replied:
Delete it from both CMS and PT. If we want to add back user inviting we can bring it back at a project template level.

lewiscollard commented:
<details>
<summary>Issue fixed: modernise the 'Add page' screen</summary>
Added: I think the 'add page' screen could look a bit more in place. Devsign time!

![screen shot 2018-08-15 at 12 52 09](https://user-images.githubusercontent.com/4973097/44147009-12d1c244-a08a-11e8-9912-79c7ff623a18.png)

Long-standing problems with the list only being able to handle one line of text (convert this to use flexbox or grid):

![screen shot 2018-08-15 at 12 54 59](https://user-images.githubusercontent.com/4973097/44147101-80ae0598-a08a-11e8-9cfe-c5bbbcd73a53.png)

More of a CMS issue as these are being overridden in the project template, but the default 'content' and 'link' icons look even more out of place in Jet (let's get the PT overrides into the CMS):
![screen shot 2018-08-15 at 12 56 45](https://user-images.githubusercontent.com/4973097/44147139-b40fcc5a-a08a-11e8-9044-2ba697b03f2a.png)

![screen shot 2018-08-15 at 12 57 07](https://user-images.githubusercontent.com/4973097/44147149-be0bcd8a-a08a-11e8-87ec-038f4e3e9e9e.png)
</details>

Commit:
Remove overrides for link + default content icon as modernised ones are now in the CMS.

Commit:
Add larger page type icons.

AidanCurrah committed:
Remove invite user link override as it's being removed from the CMS

AidanCurrah committed:
Remove video handler functions as videos are handled by the CMS

AidanCurrah commented:
Relevant Issue for sorted-m2m: https://github.com/gregmuellegger/django-sortedm2m/issues/73

lewiscollard commented:
New bug added: Popup for adding an image on an `ImageRefField` doesn't seem to work. e.g. for a news article, next to the 'Image' field, click the 'Add' button - it takes you away from the page, and saving the image gives you a blank page.

lewiscollard commented:
<details>
<summary>Issue fixed: exception on image file change form</summary>


Added two bugs on the media file page. The exception:

```
change_form.js:5 Uncaught TypeError: $ is not a function
    at change_form.js:5
    at change_form.js:20
(anonymous) @ change_form.js:5
(anonymous) @ change_form.js:20
15:22:16.533 prepopulate_init.js:3 Uncaught TypeError: $ is not a function
    at prepopulate_init.js:3
    at prepopulate_init.js:10
```
</details>

lewiscollard commented:
<details>
<summary>Issue fixed: exception on media file change form</summary>

Also, this exception happens when you go to a non-image media file page:

```
change_form.js:5 Uncaught TypeError: $ is not a function
    at change_form.js:5
    at change_form.js:20
(anonymous) @ change_form.js:5
(anonymous) @ change_form.js:20
15:25:10.650 prepopulate_init.js:3 Uncaught TypeError: $ is not a function
    at prepopulate_init.js:3
    at prepopulate_init.js:10
```
</details>

Commit:
Remove CMS password views from urls.py - let's rely on Django password validators.

Commit:
Add AUTH_PASSWORD_VALIDATORS - let's start at min 8 length.

Commit:
Move auth pipeline stuff into the PT, as it's getting removed from the CMS.

Commit:
Remove NAVIGATION_FIELDS from news article admin - unused, and now more obviously out of place on Jet admin tabs.

AidanCurrah committed:
Update compact inline editor and sections.js to deal with the new fields. Also break out the container for new sections to prevent weird re-ordering issues.

AidanCurrah committed:
Fix wysiwyg  fields not being selectable on adding a new section and cleaned up some unneeded function calls.

AidanCurrah committed:
Catch changing section shown not registering wysiwyg

AidanCurrah committed:
Add hover sudo elements to allow you to skip to the submenu.

AidanCurrah committed:
Catch starting on a different tab, switching to sections and having a WYSIWYG on the first section.

AidanCurrah committed:
Make setting selector work again.

AidanCurrah committed:
Improve admin pages for various models to accommodate for jet's tab system.

AidanCurrah committed:
Don't register partners in historylinks as partners have no get_absolute_url method

AidanCurrah committed:
Don't show the order field on a section.

AidanCurrah committed:
Improve the usability of ForeignKeyRawIdFields

AidanCurrah committed:
Remove script to fix ForeignKeyRawIdFields as it's moved to the CMS.

Commit:
Rework section fields JS.

AidanCurrah committed:
Add templates for multi language selector.

Commit:
Override fewer things in the base admin template.

AidanCurrah committed:
Update compact inline

AidanCurrah committed:
Improve sidebar styles

AidanCurrah committed:
Remove the need to override the template for compact inlines, improve the login page, improve sidebar.

AidanCurrah committed:
Remove the iframe'd page editing as we're going to be using inline editing from now on

AidanCurrah committed:
Fix iframe'd edit box not going away on save

AidanCurrah committed:
Merge branch 'develop' into feature-JetMigration

AidanCurrah committed:
Add missing utils file

AidanCurrah committed:
Remove extra whitespace

AidanCurrah committed:
Add ordering to news categories

AidanCurrah committed:
Fix unit tests

AidanCurrah committed:
catch if we're on a page with no sidebar.

AidanCurrah committed:
Fix bad templatetag check

AidanCurrah committed:
Remove some additional suit holdovers

AidanCurrah committed:
Override the default user model by default

This makes it easier if we need to change the user model in the future.

AidanCurrah committed:
Remove unused osm-cms-redirects package

The redirects app is now integrated into the project template

AidanCurrah committed:
Add the setting to set the default online status of objects to `base.py`

Even though it defaults to True, this should help us not forget this setting exists

AidanCurrah committed:
Add missing imports

Commit:
Remove overlay for edit bar (it doesn't play nice with Jet) and change "You are logged in as" to a more useful "Log out".

Commit:
Remove popup var from edit bar link.

Commit:
Remove now-unused CSS.

AidanCurrah committed:
Add osm-jet and related settings.

Also add a local dashboard with the selected default widgets.

daniel-clayton committed:
Merge branch 'develop' into feature-JetMigration

AidanCurrah committed:
Use the OSM Jet CompactInline override

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py:
```diff
@@ -0,0 +1,21 @@
+        '''
+            Add dashboard widgets for the site here.
+
+            Any widgets you add under `self.avaliable_children` will be avaliable in the widget selector.
```
available* both times

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/models.py:
```diff
@@ -1,10 +1,116 @@
 from ...utils.models import HasLinkMixin
 
 
+class UserManager(BaseUserManager):
```
I would really like this to be broken out into a separate PR, and definitely not in the `site` app.

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py:
```diff
@@ -0,0 +1,21 @@
+        self.available_children.append(modules.RecentActions)
+        self.available_children.append(modules.LinkList)
+        self.children.append(modules.RecentActions(
+            'Recent Actions',
```
Use sentence case - "Recent actions".

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/sections/admin.py:
```diff
@@ -1,22 +1,27 @@
-        js = [reverse_lazy('admin_sections_js')]
+        js = [
+            reverse_lazy('admin_sections_js')
+        ] + JetCompactInline.Media.js
```
Would it not be better for `Media` to inherit from `JetCompactInline.Media`?

lewiscollard commented on {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/admin.py:
```diff
@@ -68,7 +67,7 @@ def get_form(self, request, obj=None, **kwargs):
 
 @admin.register(CareerLocation)
-class LocationAdmin(VersionAdmin):
+class LocationAdmin(admin.ModelAdmin):
```
This should continue to inherit from VersionAdmin.

Commit:
Restore inheritance from VersionAdmin to LocationAdmin in careers.

Commit:
Fix spelling in a comment.

Commit:
Change 'Recent Actions' to 'Recent actions'.

Commit:
Remove unnecessary inheritance from `object`.

Commit:
Remove unnecessary `pass` statements.

Commit:
Fix CompactInline -> JetCompactInline.

Commit:
Remove site-local User model (it's a good idea, but should not be in this PR, and should not be in the "site" app).
